### PR TITLE
Run all compressed tiffs through libtiff

### LINKED ISSUE #2839: TIFF_LZW decompression is incorrect for  16bpc image
Pillow was refusing to open this TIFF in version 4.2.1. In version 4.3.0, it loads it, but then saving it in any format including TIFF produces a silent corruption. Python 2.7.14 in both cases.

This is the reproducing script:
```
from PIL import Image                                                                                                                      

# https://s3-external-1.amazonaws.com/prod-elife-published/articles/32330/elife-32330-fig1-v1.tif 
src_fp = 'elife-32330-fig1-v1.tif'
im = Image.open(src_fp)
target_fp = open('result.tif', 'w')
im.save(target_fp)
```

In 4.2.1:
```
  File "reproducing.py", line 5, in <module>
    im = Image.open(src_fp)
  File "/opt/loris/venv/local/lib/python2.7/site-packages/PIL/Image.py", line 2519, in open
    % (filename if filename else fp))
IOError: cannot identify image file 'elife-32330-fig1-v1.tif'
```

In 4.3.0, no output, file gets stored and corrupted.
[elife-32330-fig1-v1.tif.gz](https://github.com/python-pillow/Pillow/files/1450479/elife-32330-fig1-v1.tif.gz)
[result.tif.gz](https://github.com/python-pillow/Pillow/files/1450480/result.tif.gz)
converting to PNG

I tried to track how the `Image` class changed from 4.2.1 to 4.3.0, but there were too many changes to identify the source. I don't expect it to start to work, but to continue throwing an exception if it can't be read safely.

### PR DESCRIPTION
**Pull request by wiredfool**
Supercedes #2882 
Fixes #2839, #2503, #1270 

Changes proposed in this pull request:

 * All tiff reads of compressed tiffs run through libtiff
 * Test helper to show differences on failures
 * More end to end tiff tests

### CODE REVIEW

**Comment by wiredfool** [comment]
release notes #2832

**Comment by wiredfool** [comment]
Ok, I think this is good. I've removed the tiff_lzw decoder, as it's only used for Tiffs. The packbits decoder is still used in the PsdImagePlugin, so it remains in the tree. Jpeg is obviously used elsewhere, but support in TiffImagePlugin has been removed, as it's got colorspace issues.

File: docs/handbook/image-file-formats.rst [hunk 1]
```diff
@@ -525,13 +525,18 @@ For more information about the SPIDER image processing package, see the
...
+libtiff and its headers installed, PIL can read and write many kinds
+of compressed TIFF files. If not, PIL will only read and write
+uncompressed files.
 
-If you have libtiff and its headers installed, PIL can read and write many more
-kinds of compressed TIFF files. If not, PIL will always write uncompressed
-files.
+.. note::
+
+    Beginning in version 4.4.0, Pillow requires libtiff to read or
```
Comment (homm) [inline docs/handbook/image-file-formats.rst]: version 5.0

Comment (wiredfool) [reply docs/handbook/image-file-formats.rst]: We're going to have to do a blanket 4.4/5.0 replace.

File: PIL/TiffImagePlugin.py [hunk 2]
```diff
@@ -1285,8 +1265,13 @@ def _setup(self):
                 # we're expecting image byte order. So, if the rawmode
                 # contains I;16, we need to convert from native to image
                 # byte order.
-                if self.mode in ('I;16B', 'I;16') and 'I;16' in rawmode:
+                if rawmode == 'I;16':
                     rawmode = 'I;16N'
+                if '16B' in rawmode:
```
[the remark this answers is not in the document]
Comment (wiredfool) [reply PIL/TiffImagePlugin.py]: I don't think that's actually necessary, as there's no other use of `16B` without the `;`

[the remark this answers is not in the document]
Comment (homm) [reply PIL/TiffImagePlugin.py]: There is no use other `16B`, but this more explicitly tells where you are expecting to find this and which part of mode this should be.

**Pull request merged**